src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -630,7 +630,7 @@ var ReactCompositeComponentMixin = {
     if (this._pendingElement != null) {
       ReactReconciler.receiveComponent(
         this,
-        this._pendingElement || this._currentElement,
+        this._pendingElement,
```

Comment:
There's a gate on 630 `if (this._pendingElement != null) {`, so it can't be null there, but it might be undefined?